lib/wasi/src/state/mod.rs

```diff
@@ -128,10 +128,18 @@ pub struct Fd {
 }
 
 impl Fd {
+    /// This FD can be used with read system calls.
     pub const READ: u16 = 1;
+    /// This FD can be used with write system calls.
     pub const WRITE: u16 = 2;
+    /// This FD can append in write system calls. Note that append
```

Comment:
"Note that append permissions require the write permission."

What happens if I open for read and append but not write? Does the open fail? Does it automatically set write for me?